src/Microsoft.Data.SqlClient/src/Microsoft/Data/ProviderBase/DbConnectionPoolCounters.netfx.cs

```diff
@@ -2,6 +2,8 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+#if NETFRAMEWORK
+
```

Comment:
The performance counters and event counters are a little irritating to merge, and aren't always incremented or decremented in precisely the same places. Rather than moving DbConnectionPoolCounters (and the event counters equivalent) into the shared project, I'd rather we look at the OpenTelemetry metrics issue and replace them both with a single shared implementation.

Reply:
Although we've got someone exploring the OpenTelemetry implementation, I think it will be a ways off compared to bringing together the codebases. Can we just move the performance counters over for now and when otel is going, we can remove it?

Reply:
If OpenTelemetry's not going to be available for a while, I've got no objections. In that case, can you add the file to both projects to keep the approach consistent please?

Reply:
You mean add DbConnectionPoolCounters to both netfx and netcore projects? Netcore didn't have it at the time of doing this merge, and I don't really want to make big functional changes like that during these merges. Unless Unless I'm missing something...

Reply:
Not exactly - I'm asking for the .cs file to be included in both projects (so the approach for framework-specific classes is consistent with the .NET 6.0-only types) and for the contents of the file to be wrapped in `#if NETFRAMEWORK` as they already are in this PR.

Reply:
🤔 I don't see a precedent for including netfx/netcore files in the opposite project... at least not by searching for netfx in the netcore csproj, or vice versa. DbConnectionPoolCounters is in the common project now and is wrapped with `#if NETFRAMEWORK`

Reply:
> DbConnectionPoolCounters is in the common project now and is wrapped with #if NETFRAMEWORK

Thanks, I'd forgotten about the common project. If it's in there, we probably don't need it in the netcore project.

It's academic with the common project in place, but the precedent I had in mind was the set of files containing the [`NetSecurityNative`](https://github.com/dotnet/SqlClient/blob/main/src/Microsoft.Data.SqlClient/netcore/src/Common/Interop/Unix/System.Net.Security.Native/Interop.NetSecurityNative.cs) PInvoke and associated types; these are included in both target frameworks for the netcore project, even though they're only used in one of them.

Reply:
Forgotten about the common project 😲 I figured that's the whole point of this work 😄